test/integration/circuit_breakers_integration_test.cc

```diff
@@ -0,0 +1,65 @@
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+
+#include "test/integration/http_protocol_integration.h"
+
+namespace Envoy {
+namespace {
+
+class CircuitBreakersIntegrationTest : public HttpProtocolIntegrationTest {
+public:
+  void initialize() override { HttpProtocolIntegrationTest::initialize(); }
+};
+
+INSTANTIATE_TEST_SUITE_P(Protocols, CircuitBreakersIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams()),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+TEST_P(CircuitBreakersIntegrationTest, CircuitBreakersWithOutlierDetection) {
```

Comment:
Can you add some comment on what this test is testing as it's a little complicated?